internal/resources/grafana/resource_data_source_config_lbac_rules.go

```diff
@@ -0,0 +1,221 @@
+	}
+
+	_, err = client.Enterprise.UpdateTeamLBACRulesAPI(params)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to create LBAC rules", err.Error())
+		return
+	}
+
+	data.ID = types.StringValue(data.DatasourceUID.ValueString())
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *resourceDataSourceConfigLBACRules) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data resourceDataSourceConfigLBACRulesModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	datasourceUID := data.DatasourceUID.ValueString()
+	client := r.client.GrafanaAPI
+
+	getResp, err := client.Enterprise.GetTeamLBACRulesAPI(datasourceUID)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to get LBAC rules", err.Error())
+		return
+	}
+
+	rulesMap := make(map[string][]string)
+	for _, rule := range getResp.Payload.Rules {
+		rulesMap[rule.TeamUID] = rule.Rules
+	}
+
+	rulesJSON, err := json.Marshal(rulesMap)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to encode rules", err.Error())
+		return
+	}
+
+	data = resourceDataSourceConfigLBACRulesModel{
+		ID:            types.StringValue(datasourceUID),
+		DatasourceUID: types.StringValue(datasourceUID),
+		Rules:         types.StringValue(string(rulesJSON)),
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *resourceDataSourceConfigLBACRules) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data resourceDataSourceConfigLBACRulesModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	rulesMap := make(map[string][]string)
+	err := json.Unmarshal([]byte(data.Rules.ValueString()), &rulesMap)
+	if err != nil {
+		resp.Diagnostics.AddError("Invalid rules JSON", fmt.Sprintf("Failed to parse rules: %v", err))
+		return
+	}
+
+	apiRules := make([]*models.TeamLBACRule, 0, len(rulesMap))
+	for teamUID, rules := range rulesMap {
+		apiRules = append(apiRules, &models.TeamLBACRule{
+			TeamID:  "",
+			TeamUID: teamUID,
+			Rules:   rules,
+		})
+	}
+	datasourceUID := data.DatasourceUID.ValueString()
+	client := r.client.GrafanaAPI
+
+	params := &enterprise.UpdateTeamLBACRulesAPIParams{
+		Context: ctx,
+		UID:     datasourceUID,
+		Body:    &models.UpdateTeamLBACCommand{Rules: apiRules},
+	}
+
+	_, err = client.Enterprise.UpdateTeamLBACRulesAPI(params)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to update LBAC rules", err.Error())
+		return
+	}
+
+	data.ID = types.StringValue(datasourceUID)
+	data.DatasourceUID = types.StringValue(datasourceUID)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *resourceDataSourceConfigLBACRules) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	resp.Diagnostics.AddWarning("Operation not supported", "Delete operation is not supported for LBAC rules")
```

Comment:
We support removing LBAC rules on Grafana side, right? I don't think we can have a resource that can't be deleted.

Reply:
------------
basically terraform states that it is deleted, but it was not deleted on the grafana side of things. il implement the delete